### PR TITLE
Relax history/bookmarks DB version migration requirements

### LINKED ISSUE #2402: Relax bookmarks database version requirement during migration
Currently we require v39 as the db version for the bookmarks migration. In 39, bookmarks schema received two new fields: LOCAL_VERSION and SYNC_VERSION, none of which should be relevant for the migration.

Before that, I kind of loose the thread of the migrations, doesn't seem like bookmarks schema changed since v17 (https://bugzilla.mozilla.org/show_bug.cgi?id=961238), and likely even sometime before that.

v17 came out in Fennec 30. Let's see if we relax our migration requirement to db version 17. 

Similar issue for history: https://github.com/mozilla/application-services/issues/2401

┆Issue is synchronized with this [Jira Story](https://jira.mozilla.com/browse/SYNC-541)
┆Epic: Fenix Migration
┆Sprint: SYNC - end 2020-01-17

### PR DESCRIPTION
**Pull request by grigoryk**
Closes #2402 and #2401 (also, see those issues for version details).

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `automation/all_tests.sh` runs to completion and produces no failures
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: components/places/src/import/fennec/bookmarks.rs [hunk 1]
```diff
@@ -18,8 +18,11 @@ use sql_support::ConnExt;
 use std::time::Instant;
 use url::Url;
 
-// From https://searchfox.org/mozilla-central/rev/597a69c70a5cce6f42f159eb54ad1ef6745f5432/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java#73.
-const FENNEC_DB_VERSION: i64 = 39;
+// Fennec's bookmarks schema didn't meaningfully change since 17, so this could go as low as that version.
+// However, 23 was quite easy to obtain test databases for, and it shipped with quite an old ESR version (38).
+// The only difference between the latets (39) and this DB version is addition of LOCAL_VERSION and SYNC_VERSION
```
Comment (rfk) [inline components/places/src/import/fennec/bookmarks.rs:23]: nit: "latets" -> "latest"

**Comment by rfk** [comment]
If you feel inspired to do so, please also add a brief changelog entry (although this is easy enough to add after-the-fact).

File: components/places/src/import/fennec/history.rs [hunk 2]
```diff
@@ -13,8 +13,9 @@ use sql_support::ConnExt;
 use std::time::Instant;
 use url::Url;
 
-// From https://searchfox.org/mozilla-central/rev/597a69c70a5cce6f42f159eb54ad1ef6745f5432/mobile/android/base/java/org/mozilla/gecko/db/BrowserDatabaseHelper.java#73.
-const FENNEC_DB_VERSION: i64 = 39;
+// Fennec's history schema didn't meaningfully change since 34, so this could go as low as that version.
+// However, 36 was quite easy to obtain test databases for, and it shipped with quite an old ESR version (52).
+const FENNEC_DB_VERSION: i64 = 34;
```
Comment (cadeyrn) [inline components/places/src/import/fennec/history.rs:18]: The comment sounds like 36 is used but 34 is used. Is this intended?

Comment (grigoryk) [reply components/places/src/import/fennec/history.rs:18]: Ah, good catch - comment is wrong here, I switched it around a few times.

**Pull request merged**